code/modules/events/traders.dm

```diff
@@ -8,41 +8,73 @@ GLOBAL_LIST_INIT(unused_trade_stations, list("sol"))
 	var/station = null
 	var/spawn_count = 2
 	var/list/trader_objectives = list()
+	var/trader_type
+	var/trader_location
+	var/dock_site
+	var/trader_species
+	var/trader_outfit
+	var/trader_minor_special
+	var/trader_major_special
 
 /datum/event/traders/setup()
 	if(GLOB.unused_trade_stations.len)
 		station = pick_n_take(GLOB.unused_trade_stations)
 
 /datum/event/traders/fake_announce()
 	. = TRUE
+	var/datum/event/traders/T = pick(/datum/event/traders/sol,
+								/datum/event/traders/cyber,
+								/datum/event/traders/commie,
+								/datum/event/traders/unathi,
+								/datum/event/traders/vulp,
+								/datum/event/traders/ipc,
+								/datum/event/traders/vox,
+								/datum/event/traders/skrell,
+								/datum/event/traders/grey,
+								/datum/event/traders/nian)
+
 	if(SSsecurity_level.get_current_level_as_number() >= SEC_LEVEL_RED)
-		GLOB.minor_announcement.Announce("A trading shuttle from Jupiter Station has been denied docking permission due to the heightened security alert aboard [station_name()].", "Trader Shuttle Docking Request Refused")
+		GLOB.minor_announcement.Announce("A trading shuttle from [T.trader_location] has been denied docking permission due to the heightened security alert aboard [station_name()].", "Trader Shuttle Docking Request Refused")
 		return
-	GLOB.minor_announcement.Announce("A trading shuttle from Jupiter Station has been granted docking permission at [station_name()] arrivals port 4.", "Trader Shuttle Docking Request Accepted")
+	GLOB.minor_announcement.Announce("A trading shuttle from [T.trader_location] has been granted docking permission at [station_name()] arrivals port 4.", "Trader Shuttle Docking Request Accepted")
 
 
 /datum/event/traders/start()
 	if(!station) // If there are no unused stations, just no.
 		return
+
+	var/datum/event/traders/T = pick(/datum/event/traders/sol,
```

Comment:
This shouldnt be making subtypes of itself, if you're not careful, it could lead to infinite recursion. These should be non-event datums that you store this data on
```
/datum/trader_type/sol
	trader_type = "Trans-Solar Federation"
	trader_location = "Kayani Station"
	dock_site = "Kayani Station"
	trader_species = /mob/living/carbon/human
	trader_outfit = /datum/outfit/admin/trader/sol
	trader_minor_special = /obj/effect/spawner/lootdrop/trade_sol/federation_minor
	trader_major_special = /obj/effect/spawner/lootdrop/trade_sol/federation_major
```
You can use this exactly like how you've been using `T.` Just don't forget to `qdel(T)` once you are finished with the datum